cypress/README.md

```diff
@@ -0,0 +1,603 @@
+    (property: {
+        host: number
+    }) => {
+        describe(`Check example apps`, () => {
+            beforeEach(() => {
+                basePage.openLocalhost(host)
+            })
+
+            it(`User should be created & added`, () => {
+                basePage.clickElementBySelector({
+                    selector: baseSelectors.createButton
+                })
+                basePage.checkUserCreated(true)
+            })
+
+            it(`User should be deleted`, () => {
+                basePage.clickElementBySelector({
+                    selector: baseSelectors.deleteButton
+                })
+                basePage.checkUserDeleted(true)
+            })
+        })
+    })
+```
+
+It will be run your testCases on every host, as example on 3001 and 3002. So you need to skip delete user on 3001 host and creating user on 3002. Let's do this!
+
+First of all you need to add custom command to `cypress/support/commands.ts`:
+
+```typescript
+declare global {
+    namespace Cypress {
+        interface Chainable {
+            skipWhen(condition: boolean): void;
+        }
+    }
+}
+
+Cypress.Commands.add('skipWhen', function (condition) {
+    if (condition) {
+        this.skip()
+    }
+})
+```
+
+Then added it to `cypress/common/base.ts` file 'cause we are going to use it global:
+
+```typescript
+public skipTestByCondition(condition: any): void {
+    cy.skipWhen(condition)
+}
+```
+
+And finnaly use it in our testCase
+
+```typescript
+const appData = [
+    {
+        host: 3001
+    },
+    {
+        host: 3002
+    }
+]
+
+appsData.forEach(
+    (property: {
+        host: number
+    }) => {
+        describe(`Check example apps`, () => {
+            beforeEach(() => {
+                basePage.openLocalhost(host)
+            })
+
+            it(`User should be created & added`, () => {
+                basePage.skipTestByCondition(property.host === 3002)
+                basePage.clickElementBySelector({
+                    selector: baseSelectors.createButton
+                })
+                basePage.checkUserCreated(true)
+            })
+
+            it(`User should be deleted`, () => {
+                basePage.skipTestByCondition(property.host === 3001)
+                basePage.clickElementBySelector({
+                    selector: baseSelectors.deleteButton
+                })
+                basePage.checkUserDeleted(true)
+            })
+        })
+    })
+```
+
+`skipTestByCondition(ocndition)` will skipped test if their app host will be 3001 or 3002 respectivelly. Isn't it cool? 🔥🔥🔥
+### Use objects
+
+##### When more that one app will be tested and this apps almost similar
+
+For example:
+
+App1
+![App1](https://i.ibb.co/26kNLXQ/Screenshot-2023-01-11-at-14-59-59.png)
+App2
+![App2](https://i.ibb.co/tZGn3F0/Screenshot-2023-01-11-at-15-00-03.png)
+
+We have two apps `app1` and `app2` they are simillary. So we can create one object for both apps. And then use it in our tests.
```

Comment:
A little bit confusing too. First you introduce skipWhen functionality and show it on commonChecks approach and only later you introduce commonChecks itself. Better to change blocks positions

Reply:
Fixed